File: yh_star/src/yh_star_pub.cpp
# include "ros/ros.h"
# include "yh_star/YhStarMsg.h"

int main(int argc, char** argv)
{
    ros::init(argc,argv, "yh_star_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<yh_star::YhStarMsg>("yh_star_topic", 5);

    ros::Rate loop_rate(2);

    yh_star::YhStarMsg msg;
    msg.data = 1;

    while(ros::ok())
    {
        msg.stamp = ros::Time::now();
        pub.publish(msg);
        msg.data++;
        loop_rate.sleep();
    }

    return 0;
}

File: yh_star/scripts/yh_star_pub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import rospy
from yh_star.msg import YhStarMsg

def star_publisher():
    rospy.init_node("yh_star_pub")
    pub = rospy.Publisher("yh_star_topic", YhStarMsg, queue_size=10)

    loop_rate = rospy.Rate(2)

    msg = YhStarMsg()
    msg.data = 1

    while not rospy.is_shutdown():
        msg.stamp = rospy.Time.now()
        pub.publish(msg)
        msg.data += 1
        loop_rate.sleep()

if __name__ == "__main__":
    star_publisher()